tests/model.rs

```diff
@@ -0,0 +1,18 @@
+extern crate sparse_bitfield;
+#[macro_use]
+extern crate proptest;
+
+use sparse_bitfield::{Bitfield, Change};
+
+fn model(bit: usize, page_sz: usize) {
+  let mut bits = Bitfield::new(page_sz);
+  assert_eq!(bits.set(bit, true), Change::Changed);
+  assert_eq!(bits.get(bit), true);
+}
+
+proptest! {
+  #[test]
+  fn doesnt_crash(bit in 0usize..1_000_000, page_sz_exponent in 0usize..30) {
```

Comment:
Sorry for newb question. But how does this work? The parameters in a function have a range and proptest uses this to invoke it for all pairs somehow?

Reply:
Ok cool. But is it the `proptest!` macro doing this or this is a feature of rust itself?

Reply:
proptest randomly chooses a value within a `Range`. in rust, `n..m` is shorthand for an instance of the `std::ops::Range` type.